src/plugins/query_enhancements/server/connections/resource_manager_service.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BaseConnectionManager } from './managers/base_connection_manager';
+
+/**
+ * @experimental this class is experimental and might change in future releases.
+ */
+class ResourceManagerService {
+  private readonly managers: Map<string, BaseConnectionManager>;
+  constructor() {
+    this.managers = new Map();
+  }
+
+  register(dataConnectionType: string, manager: BaseConnectionManager) {
+    if (this.managers.get(dataConnectionType) !== undefined) {
+      throw new Error(
+        `Manager for dataConnectionType ${dataConnectionType} is already registered. Unable to register another manager.`
+      );
+    }
+    this.managers.set(dataConnectionType, manager);
+  }
+
+  getManager(dataConnectionType: string): BaseConnectionManager | undefined {
+    return this.managers.get(dataConnectionType);
+  }
+}
+
+// export as singleton
+export const resourceManagerService = new ResourceManagerService();
```

Comment:
nit: the way core does this is via getters and setters. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/services.ts#L37 

just a heads up placing this in query enhancements could be a little risky. query enhancements is optionally enabled. technically we can disable this plugin and now the resource manager service will no longer exist. 

typically we try to rely on key plugins with registration points that if disabled OSD does not function at all.

perhaps that could be a follow up. more so in protection of consumer plugins not relying on an optional plugin